code/__DEFINES/maths.dm

```diff
@@ -1,234 +1,199 @@
+/proc/SolveQuadratic(a, b, c)
+	ASSERT(a)
+	. = list()
+	var/d		= b*b - 4 * a * c
+	var/bottom  = 2 * a
+	if(d < 0)
+		return
+	var/root = sqrt(d)
+	. += (-b + root) / bottom
+	if(!d)
+		return
+	. += (-b - root) / bottom
+
+#define TODEGREES(radians) (radians * 57.2957795)
+
+#define TORADIANS(degrees) (degrees * 0.0174532925)
+
+// Will filter out extra rotations and negative rotations
+// E.g: 540 becomes 180. -180 becomes 180.
+#define SIMPLIFY_DEGREES(degrees) ((degrees*SIGN(degrees))%360)
+
+// min is inclusive, max is exclusive
+#define WRAP(val, min, max) (val - ((round(val - min) / (max - min)) * (max - min)))
+
+#define NORM_ROT(rot) ((((rot % 360) + (rot - round(rot, 1))) >= 0) ? ((rot % 360) + (rot - round(rot, 1))) : (((rot % 360) + (rot - round(rot, 1))) + 360))
+
+/proc/get_angle_of_incidence(face_angle, angle_in, auto_normalize = TRUE)
+	var/angle_in_s = NORM_ROT(angle_in)
+	var/face_angle_s = NORM_ROT(face_angle)
+	var/incidence = face_angle_s - angle_in_s
+	var/incidence_s = incidence
+	while(incidence_s < -90)
+		incidence_s += 180
+	while(incidence_s > 90)
+		incidence_s -= 180
+	if(auto_normalize)
+		return incidence_s
+	else
+		return incidence
+
+//A logarithm that converts an integer to a number scaled between 0 and 1 (can be tweaked to be higher).
+//Currently, this is used for hydroponics-produce sprite transforming, but could be useful for other transform functions.
+//Kept for documentation reasons
+///proc/TRANSFORMUSINGVARIABLE(input, inputmaximum, scaling_modifier = 0)
+//
+//		var/inputToDegrees = (input/inputmaximum)*180 //Converting from a 0 -> 100 scale to a 0 -> 180 scale. The 0 -> 180 scale corresponds to degrees
+//		var/size_factor = ((-cos(inputToDegrees) +1) /2) //returns a value from 0 to 1
+//
+//		return size_factor + scaling_modifier //scale mod of 0 results in a number from 0 to 1. A scale modifier of +0.5 returns 0.5 to 1.5
+#define TRANSFORM_USING_VARIABLE(input, max, scale) ( ((input/max)*180) + (-cos((input/max)*180)) )
+
+//converts a uniform distributed random number into a normal distributed one
+//since this method produces two random numbers, one is saved for subsequent calls
+//(making the cost negligble for every second call)
+//This will return +/- decimals, situated about mean with standard deviation stddev
+//68% chance that the number is within 1stddev
+//95% chance that the number is within 2stddev
+//98% chance that the number is within 3stddev...etc
+#define ACCURACY 10000
+/proc/gaussian(mean, stddev)
+	var/static/gaussian_next
+	var/R1;var/R2;var/working
+	if(gaussian_next != null)
+		R1 = gaussian_next
+		gaussian_next = null
+	else
+		do
+			R1 = rand(-ACCURACY,ACCURACY)/ACCURACY
+			R2 = rand(-ACCURACY,ACCURACY)/ACCURACY
+			working = R1*R1 + R2*R2
+		while(working >= 1 || working==0)
+		working = sqrt(-2 * log(working) / working)
+		R1 *= working
+		gaussian_next = R2 * working
+	return (mean + stddev * R1)
+#undef ACCURACY
+
+/proc/mouse_angle_from_client(client/client)
```

Comment:
```
//Refer to http://www.byond.com/docs/ref/info.html#/client/var/view for mad maths
/proc/mouse_angle_from_client(client/c)
	var/screen_loc = params2list(c.mouseParams)["screen-loc"]
	if(screen_loc)
		var/list/screen_loc_params = splittext(screen_loc, ",")
		var/list/screen_loc_X = splittext(screen_loc_params[1],":")
		var/list/screen_loc_Y = splittext(screen_loc_params[2],":")
		var/halfscreen = round((c.view * 2 + 1) * world.icon_size / 2)
		return NORM_ROT(ATAN2((text2num(screen_loc_Y[1]) * 32 + text2num(screen_loc_Y[2]) - 32) - halfscreen + c.pixel_y, (text2num(screen_loc_X[1]) * 32 + text2num(screen_loc_X[2]) - 32) - halfscreen + c.pixel_x))
```

Reply:
too little benefit for the loss of readability